components/style/properties/properties.mako.rs

```diff
@@ -1003,44 +1064,78 @@ impl UnparsedValue {
             );
 
             let mut input = ParserInput::new(&css);
-            Parser::new(&mut input).parse_entirely(|input| {
-                match self.from_shorthand {
-                    None => longhand_id.parse_value(&context, input),
-                    Some(ShorthandId::All) => {
-                        // No need to parse the 'all' shorthand as anything other than a CSS-wide
-                        // keyword, after variable substitution.
-                        Err(input.new_custom_error(SelectorParseErrorKind::UnexpectedIdent("all".into())))
+            Parser::new(&mut input)
+                .parse_entirely(|input| parse(&context, input))
+                .ok()
+        })
+    }
+
+    fn substitute_variables_from_longhand(
+        &self,
+        longhand_id: LonghandId,
+        custom_properties: Option<<&Arc<::custom_properties::CustomPropertiesMap>>,
+        quirks_mode: QuirksMode,
+    ) -> Option<PropertyDeclaration> {
+        self.substitute_variables_and_parse(
+            custom_properties,
+            quirks_mode,
+            |context, input| longhand_id.parse_value(context, input)
+        )
+    }
+
+    fn substitute_variables_from_shorthand<'i, 't, 'a>(
+        &'a self,
+        longhand_id: LonghandId,
+        custom_properties: Option<<&Arc<::custom_properties::CustomPropertiesMap>>,
+        quirks_mode: QuirksMode,
+        cache: &mut ParsedShorthandCache<'a>,
+    ) -> Option<PropertyDeclaration> {
+        // Propagate the cache when cache misses.
```

Comment:
I don't know what `Propagate the cache` means. You mean `Populate`?

Reply:
Yeah, should be populate...